volatility3/framework/plugins/windows/writecrashdump.py

```diff
@@ -0,0 +1,134 @@
+        primary = context.layers[layer_name]
+        is_pae = isinstance(primary, intel.IntelPAE)
+        is_64_bit = isinstance(primary, intel.Intel32e)
+
+        if is_64_bit:
+            crashdump_json = 'crash64'
+            dump_header_name = '_DUMP_HEADER64'
+            valid_dump_suffix = [ord('6'), ord('4')]
+        else:
+            crashdump_json = 'crash'
+            dump_header_name = '_DUMP_HEADER'
+            valid_dump_suffix = [ord('M'), ord('P')]
+
+        config_path = 'whatever'
+        crash_table_name = intermed.IntermediateSymbolTable.create(context,
+                                                                   configuration.path_join(config_path, 'symbols'),
+                                                                   '',
+                                                                   crashdump_json)
+
+        dump_header_type = context.symbol_space[crash_table_name].get_type(dump_header_name)
+        header_layer_name = context.layers.free_layer_name('header_layer')
+        header_layer = physical.BufferDataLayer(context,
+                                                configuration.path_join(config_path, 'layer'),
+                                                name = header_layer_name,
+                                                buffer = b"PAGE" * (dump_header_type.size // 4))
+        context.layers.add_layer(header_layer)
+        dump_header = context.object(dump_header_type, header_layer.name, 0)
+
+        info_config_path = configuration.path_join(config_path, 'info')
+        kdbg = info.Info.get_kdbg_structure(context, info_config_path, layer_name, symbol_table)
+
+        kuser = info.Info.get_kuser_structure(context, layer_name, symbol_table)
+        version_structure = info.Info.get_version_structure(context, layer_name, symbol_table)
+
+        dump_header.ValidDump.write([ord('D'), ord('U')] + valid_dump_suffix)
+
+        # TODO: remove the OR after #702 gets merged in.
+        dump_header.KdDebuggerDataBlock.write(kdbg.vol.offset | (0 if not is_64_bit else 0xFFFF000000000000))
+        dump_header.MajorVersion.write(version_structure.MajorVersion)
+        dump_header.MinorVersion.write(version_structure.MinorVersion)
+        dump_header.MachineImageType.write(version_structure.MachineType)
+
+        number_processors = kernel.object("unsigned int", offset=kernel.get_symbol("KeNumberProcessors").address)
+        dump_header.NumberProcessors.write(number_processors)
+
+        dump_header.DirectoryTableBase.write(primary.config['page_map_offset'])
+        if dump_header.has_member("PaeEnabled"):
+            dump_header.PaeEnabled.write(int(is_pae))
+
+        dump_header.PfnDataBase.write(kdbg.MmPfnDatabase)
+        dump_header.PsLoadedModuleList.write(kdbg.PsLoadedModuleList)
+        dump_header.PsActiveProcessHead.write(kdbg.PsActiveProcessHead)
+        dump_header.DumpType.write(1)
```

Comment:
A comment to explain the value 1 here would be a nice-to-have, but not critical  5:)

Reply:
added a comment